metaflow/package.py

```diff
@@ -75,6 +76,19 @@ def path_tuples(self):
                 addl_suffixes=self.metaflow_extensions_addl_suffixes,
             ):
                 yield path_tuple
+
+        # Any custom packages exposed via decorators
+        deco_module_paths = set()
+        for step in self._flow:
+            for deco in step.decorators:
+                for path_tuple in deco.add_to_package():
+                    file_path, _ = path_tuple
+                    # Check if the path is not duplicated as
+                    # many steps can have the same packages being imported
+                    if file_path not in deco_module_paths:
```

Comment:
I was commenting on this in the cards PR (not yet submitted) but what happens if multiple decorators want to include the same file but as different name? I am not sure whey this can happen but it seems a possibility if you expose the name (which is then ignored) to the plugins.

Reply:
Interesting point. 

Some context to why I have kept the `path_tuples` as a return type from `add_to_package`: 
1. path_tuples consists of `path,arcname` 
2. `arcname` is useful when we untar the package on a remote execution as the relative path in the `arcname` makes a difference in how imports will happen on the remote container. [Here](https://github.com/valayDave/metaflow/blob/3567a2dffaa4287b786fc226ec4b24750442d13a/metaflow/plugins/cards/card_decorator.py#L81) is an example in the Cards PR where I needed to ensure correct path in the arcname to ensure remote imports works correctly.

Now coming to your point. Yes, I don't currently check for `same file but as different name` and I can change that. I can check for both the `file_path` as well as the `arcname` by adding an object like `arcname-filepath` in `deco_module_paths`. 

Does that resolve this issue ?